.github/workflows/generate-gql-types.yml

```diff
@@ -18,41 +18,34 @@ jobs:
         node-version: [21.3.0]
 
     steps:
-      # Checkout the repository
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
+          
       - name: Install Yarn/Node ${{ matrix.node-version }}
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
           cache: 'yarn'
+
       - name: Install Dependencies
         run: yarn install --immutable
+
       - name: Generate GQL Types
         run: yarn generate
-      - name: Check for changes
-        run: git diff --exit-code -- typedefs/
-        continue-on-error: true
-
-      # Commit and push changes if any exist
-      - name: Commit changes
-        if: failure()
-        run: |
-          git config --global user.name "github-actions[bot]"
-          git config --global user.email "github-actions[bot]@users.noreply.github.com"
-          git add typedefs/
-          git commit -m "chore: update gql types"
-          git push
 
-      # Create a pull request if changes were committed
       - name: Create Pull Request
-        if: failure()
         uses: peter-evans/create-pull-request@v5
```

Comment:
Would be good to add a comment that this will only create a PR on if there's diffs.